helpers/error_handler/error_message.go

```diff
@@ -43,13 +43,17 @@ func ValidationErrorMessage(err FieldError) error {
 
 // TODO: Improve later as this feel brittle
 func DatabaseErrorMessage(err error) error {
-	pgErr := err.(*pgconn.PgError)
-
-	switch pgErr.Code {
-	case foreignKeyErrorCode:
+	pgErr, isPgErr := err.(*pgconn.PgError)
+
+	if isPgErr {
```

Comment:
What are the other cases of errors?

Reply:
As I read `gorm` source code, `gorm` provides only 2 error types which is `*errors.errorString` by gorm itself and `*pgconn.PgError` for other errors. That's why I use just only if else in this case 😅